core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/ValueStore.java

```diff
@@ -953,8 +958,10 @@ protected void deleteValueToIdMappings(MemoryStack stack, long txn, Collection<L
 			for (Long id : ids) {
 				idVal.mv_data(id2data(idBb.clear(), id).flip());
 				// id must not have a reference count and must have an existing value
-				if (mdb_get(writeTxn, refCountsDbi, idVal, ignoreVal) != 0 &&
-						mdb_get(txn, dbi, idVal, dataVal) == 0) {
+				int a = E(mdb_get(writeTxn, refCountsDbi, idVal, ignoreVal)); // this is where I get MDB_BAD_TXN
```

Comment:
@kenwenzel I fairly consistently get MDB_BAD_TXN here

Reply:
I think it could be due to a previous map full. Can you try to move `resizeMap` into the inner loop of `gcIds`?